site/src/pages/docs/client-factory-builder.mdx

```diff
@@ -0,0 +1,134 @@
+clients which were spawned from the <type://ClientFactory>, will also be closed.
+Therefore, you will generally avoid closing the <type://ClientFactory>, except for perhaps in a shutdown hook,
+to avoid losing connections to the other servers.
+For unit tests where a <type://ClientFactory> is only used for a single test or test class,
+it is appropriate to close it when done using it.
+It might be good suggestion keeping your <type://ClientFactory> lifecycle same as application's lifecycle.
+
+## Specifying Netty `ChannelOption`s
+
+You can also specify Netty [`ChannelOption`](https://netty.io/4.1/api/io/netty/channel/ChannelOption.html)
+via <type://ClientFactoryBuilder#channelOption(ChannelOption,T)>:
+
+```java
+final ClientFactory factory =
+        ClientFactory.builder()
+                     // Set the size of send socket buffer as 4096 bytes
+                     .channelOption(ChannelOption.SO_SNDBUF, 4096)
+                     .build();
+// Do something with factory
+WebClient.builder().factory(factory).build();
+```
+
+## Specifying different `EventLoop` threads for different endpoints
+
+By default, Armeria keeps at most 1 connection per endpoint (host and port pair) in case of using HTTP/2,
+which is desirable for typical HTTP/2 services. Armeria achieves this by assigning one `EventLoop` per endpoint
+from the <type://ServerConfig#workerGroup()>. However, there's a limit to the requests that
+a single connection(or `EventLoop` which is a thread) can handle, so you might want to increase
+the number of connections(or `EventLoop` threads) for certain or all endpoints.
+
+You can do this via <type://ClientFactoryBuilder#maxNumEventLoopsPerEndpoint(int)>:
+
+The <type://ClientFactoryBuilder> class has a feature to set the number of `EventLoop` threads using
+`maxNumEventLoopsPerEndpoint` and `maxNumEventLoopsFunction`. The former parameter will set the number of `EventLoop`
+threads for each endpoint (This parameter is used for HTTP/2.
+In HTTP/1, use `maxNumEventLoopsPerHttp1Endpoint` instead.).
+It indicates that if you change this value `N`,
+then Client instances that are generated by the factory will use distinct `EventLoop` threads per endpoint.
+
+```java
+ClientFactory factory =
+         ClientFactory.builder()
+                      // a client will use maximum 5 EventLoops from the worker group per endpoint
+                      .maxNumEventLoopsPerEndpoint(5)
+                      .build();
+// Do something using factory
+WebClient.builder().factory(factory).build();
+
+ClientFactory factory =
+        ClientFactory.builder()
+                     // or you can just use Integer.MAX_VALUE to use all EventLoops
+                     .maxNumEventLoopsPerEndpoint(Integer.MAX_VALUE)
+                     .build();
+// Do something using factory
+WebClient.builder().factory(factory).build();
+```
+
+The client might need to send many requests to a certain endpoint, while not bothering other endpoints.
```

Comment:
```suggestion
The client might need to send many requests to a certain endpoint, while sending small number of requests to others.
```
It's not bothering other endpoints at all. 😄  Just assigned more EventLoops than it needs.

Reply:
I see :)